compiler/noirc_frontend/src/parser/parser/traits.rs

```diff
@@ -203,7 +203,7 @@ fn trait_implementation_body() -> impl NoirParser<Vec<Documented<TraitImplItem>>
         .map(|(name, alias)| TraitImplItemKind::Type { name, alias });
 
     let let_statement = let_statement(expression()).then_ignore(just(Token::Semicolon)).try_map(
-        |((pattern, typ), expr), span| match pattern {
+        |(_attributes, pattern, typ, expr), span| match pattern {
```

Comment:
Please throw an error on unexpected attributes here

Reply:
My thinking was that we'll probably eventually want to support attributes here. Or, put another way, adding attributes here is fine, it's just that this particular item doesn't support any of the existing attributes.

Reply:
My concern is that any attributes added here are ignored without warning. Since metaprogramming is only partially documented, a user could add an attribute here that silently fails.